### PR TITLE
Revert "Report correct IP & port in local mode"

### PR DESCRIPTION
**Pull request by penalosa**
Reverts cloudflare/workers-sdk#3337, since it's breaking e2e tests

### CODE REVIEW

**Review by RamIdeas — APPROVED** [review]

**Comment by GregBrimble** [comment]
Is the feature itself broken? Or does this PR now break the feature?

We shouldn't merge this if it's solely about fixing the e2e tests at the expense of the feature.

**Comment by penalosa** [comment]
@GregBrimble This reverts a PR I merged a couple of days ago. That PR fixed a bug, but it turns out that the bug-fix broke e2e tests. I'm reverting the PR so that main is green, and we can figure out a better way to fix the underlying bug without having releases blocked

**Comment by GregBrimble** [comment]
Ah apologies. I saw the other PR was created a month ago, but skimmed over the fact that it was only merged a few days ago. I assumed it was an old change.

**Pull request merged**
